### PR TITLE
Adds the crates.io ecosystem

### PR DESCRIPTION
**Pull request by another-rex**
Also updates the analyse docker file so that the go dependencies are cached before copying over the source file which changes regularly

### CODE REVIEW

File: sandboxes/cratesio/analyze.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    except subprocess.CalledProcessError as e:
+      print('Failed to install:')
+      print(e.output.decode())
+      # Always raise.
+      # Install failing is either an interesting issue, or an opportunity to
+      # improve the analysis.
+      raise
+
+
+def importPkg(package):
```
Comment (calebbrown) [inline sandboxes/cratesio/analyze.py:44]: Feel free to delete everything in here related to "import".

File: cmd/analyze/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,12 @@
 FROM golang@sha256:3c4de86eec9cbc619cdd72424abd88326ffcf5d813a8338a7743c55e5898734f as build
 RUN apt-get update && apt-get install -y libpcap-dev
 WORKDIR /src
+
+# First cache the dependencies to avoid downloading again on code change
```
Comment (calebbrown) [inline cmd/analyze/Dockerfile:5]: Great idea!

**Review by calebbrown — COMMENTED** [review]
Maybe dropping the "io" and just calling it "Crates" would be cleaner?

**Comment by calebbrown** [comment]
Thanks Rex for implementing this - I'm super interested if it'll show anything up.

**Comment by oliverchang** [comment]
> Maybe dropping the "io" and just calling it "Crates" would be cleaner?

Though we already don't do this, it would be good to keep the ecosystems more aligned with OSV (https://ossf.github.io/osv-schema/#affectedpackage-field), where this is "crates.io".

**Comment by another-rex** [comment]
> Maybe dropping the "io" and just calling it "Crates" would be cleaner?

> Though we already don't do this, it would be good to keep the ecosystems more aligned with OSV (https://ossf.github.io/osv-schema/#affectedpackage-field), where this is "crates.io".

I changed the docker container and folder names to crates.io. Can't really change the go names to have a dot in it. Maybe dropping the io part in just the go variable/type names makes sense, while keeping everything else crates.io

File: internal/pkgecosystem/crates.io.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package pkgecosystem
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type cratesIoJSON struct {
```
Comment (calebbrown) [inline internal/pkgecosystem/crates.io.go:9]: So in Go's style guide it says that acronyms need to be capitalised.

`.io` stands for "British Indian Ocean Territory" or "input/output", so this code should technically be capitalized `cargoIOJSON` (and `getCreatesIOLatest`, etc)

Its probably cleaner in this context to drop the `Io` as you suggest elsewhere than try and apply this style.

Comment (another-rex) [reply internal/pkgecosystem/crates.io.go:9]: Good point, it is now renamed to just `Crates`.

Comment (calebbrown) [reply internal/pkgecosystem/crates.io.go:9]: Thanks! I see you renamed `cratesPkgManager` below, but it looks like you may have missed a commit for the others instances of `cratesIo`.

Comment (another-rex) [reply internal/pkgecosystem/crates.io.go]: Ah my bad, forgot to press save in vscode, should be all renamed now.

**Review by calebbrown — APPROVED** [review]
Thanks Rex!

**Comment by calebbrown** [comment]
See #33

**Pull request merged**